src/services/submissionService.ts

```diff
@@ -65,28 +100,35 @@ export async function getSubmissions(filters: dbSchema.SubmissionSearchFilters =
     }
 }
 
-function convertToDbSubmission(submission: gqlType.Submission): 
+function convertToDbSubmission(submission: gqlType.SubmissionInput): 
     dbSchema.Submission {
     return {
         ...submission,
+        id: '',
+        isUnderReview: true,
+        isApproved: false,
+        isRejected: false,
+        createdDate: new Date().toISOString(),
+        updatedDate: new Date().toISOString(),
         spokenLanguages: submission.spokenLanguages
             .filter(lang => lang !== null) as dbSchema.SpokenLanguage[]
     }
 }
 
-export const addSubmission = async (submission: gqlType.Submission): 
+export const addSubmission = async (submissionInput: gqlType.SubmissionInput): 
     Promise<dbSchema.Submission> => {
     try {
-        const dbSubmission = convertToDbSubmission(submission)
+        const dbSubmission = convertToDbSubmission(submissionInput)
 
-        const db = getFirestore()
-        const submissionRef = db.collection('submissions')
+        const submissionRef = dbInstance.collection('submissions')
 
         const docRef = await submissionRef.add(dbSubmission)
 
         const newSubmission: dbSchema.Submission = {
             ...dbSubmission,
-            id: docRef.id
+            id: docRef.id,
+            createdDate: new Date().toISOString(),
```

Comment:
I think this is actually happening too late since it's already been added to the Db. Since you already have it wired up in the conversion method, I think you can just remove this and you're good to go 👍